docs/reference/aggregations/pipeline/bucket-correlation-aggregation.asciidoc

```diff
@@ -0,0 +1,141 @@
+
+.Properties of `count_correlation`
+[%collapsible%open]
+=====
+`indicator`:::
+(Required, object)
+The indicator with which to correlate the configured `bucket_path` values.
+
+.Properties of `indicator`
+[%collapsible%open]
+=====
+`expectations`:::
+(Required, array)
+An array of numbers with which to correlate the configured `bucket_path` values. The length of this value must always equal
+the number of buckets returned by the `bucket_path`.
+
+`fractions`:::
+(Optional, array)
+An array of fractions to use when averaging and calculating variance. This should be used if the pre-calculated data and the
+`buckets_path` have known gaps. The length of `fractions`, if provided, must equal `expectations`.
+
+`doc_count`:::
+(Required, integer)
+The total number of documents that initially created the `expectations`. It's required to be greater than or equal to the sum
+of all values in the `buckets_path` as this is the originating superset of data to which the term values are correlated.
+=====
+=====
+====
+
+==== Syntax
+
+A `bucket_correlation` aggregation looks like this in isolation:
+
+[source,js]
+--------------------------------------------------
+{
+  "bucket_correlation": {
+    "buckets_path": "range_values>_count", <1>
+    "function": {
+      "count_correlation": { <2>
+        "expectations": [...],
+        "doc_count": 10000
+      }
+    }
+  }
+}
+--------------------------------------------------
+// NOTCONSOLE
+<1> The buckets containing the values to correlate against.
+<2> The correlation function definition.
+
+
+[[bucket-correlation-agg-example]]
+==== Example
+
+The following snippet correlates the individual terms in the field `service.version.keyword` with the `latency` metric. Not shown
+is the pre-calculation of the `latency` indicator values, which was done utilizing the
+<<search-aggregations-metrics-percentile-aggregation,percentiles>> aggregation.
+
+Since this example has 50 individual range buckets and 50 expectation values, they are elided for brevity.
+
+[source,js]
+-------------------------------------------------
+GET apm-7.12.0-transaction-generated/_search
+{
+  "size": 0,
+  "aggs": {
+    "field_terms": {
+      "terms": {
+        "field": "service.version.keyword", <1>
+        "size": 20
+      },
+      "aggs": {
+        "latency_range": {
+          "range": { <2>
+            "field": "transaction.duration.us",
+            "ranges": [...],
+            "keyed": true
+          }
+        },
+        "correlation": {
+          "bucket_correlation": { <3>
+            "buckets_path": "latency_range>_count",
+            "count_correlation": {
+              "indicator": {
+                 "expectations": [...],
+                 "doc_count": 20000
+               }
+            }
+          }
+        }
+      }
+    }
+  }
+}
+-------------------------------------------------
+// NOTCONSOLE
```

Comment:
We should make this a fully real example, i think. It'd be a pain to make the setup for it, but without that we can't be sure it works.

Reply:
@nik9000 there is a non-doc integration test that covers this. Do xpack doc tests even run?

I can attempt to do a set up, but its gonna take a bit to generate data and write out the 50 ranges.

Reply:
Its important that we test these so they don't break eventually. I can't tell you the number of times I've broken stuff in the docs without noticing it. I mean, since we made the tests I can tell you its much rarer, but I still can't tell you.

You can totally use the `setup` stuff in `docs/build.gradle` - over there you can write for loops and stuff to emit values.

If we have a response on the page we should assert that it came from a request on the page - but its totally ok to use stuff like `...` and `filter_path` to shrink it. No one is going to read a huge response anyway.....